Neos.Flow/Classes/Mvc/Routing/Routes.php

```diff
@@ -16,33 +18,47 @@ final class Routes implements \IteratorAggregate
     private readonly array $routes;
 
     private function __construct(
-        array $routes
+        Route ...$routes
     ) {
         $this->routes = $routes;
     }
 
-    public static function fromArray(array $routes): self
+    public static function fromConfiguration(array $configuration): self
     {
-        return new self($routes);
+        $routes = self::empty();
+        $routesWithHttpMethodConstraints = [];
+        foreach ($configuration as $routeConfiguration) {
+            $route = Route::fromConfiguration($routeConfiguration);
+
+            $uriPattern = $route->getUriPattern();
+
+            if (isset($routeConfiguration['httpMethods'])) {
+                if (isset($routesWithHttpMethodConstraints[$uriPattern]) && $routesWithHttpMethodConstraints[$uriPattern] === false) {
+                    throw new InvalidRouteSetupException(sprintf('There are multiple routes with the uriPattern "%s" and "httpMethods" option set. Please specify accepted HTTP methods for all of these, or adjust the uriPattern', $uriPattern), 1365678427);
+                }
+                $routesWithHttpMethodConstraints[$uriPattern] = true;
+                $route->setHttpMethods($routeConfiguration['httpMethods']);
+            } else {
+                if (isset($routesWithHttpMethodConstraints[$uriPattern]) && $routesWithHttpMethodConstraints[$uriPattern] === true) {
+                    throw new InvalidRouteSetupException(sprintf('There are multiple routes with the uriPattern "%s" and "httpMethods" option set. Please specify accepted HTTP methods for all of these, or adjust the uriPattern', $uriPattern), 1365678432);
+                }
+                $routesWithHttpMethodConstraints[$uriPattern] = false;
+            }
+            $routes->append($route);
```

Comment:
this cant work right? as this is immutable ;)

either `$routes = $routes->append($route);`

or create a separate array and do `new self(...$routes)` at the end

Reply:
True